lib/item.rb

```diff
@@ -24,11 +26,22 @@ def invoice_items
   end
 
   def merchant
-    repository.find_merchant(id)
+    repository.find_merchant(merchant_id)
   end
 
   def best_day
     maximum_item = invoice_items.max_by { |invoice_item| invoice_item.quantity }
     maximum_item.invoice.created_at
   end
+
+  def revenue
+    successful_invoice_item = invoice_items.select do |inv_item|
+      inv_item.invoice.transactions.each do |tran|
+        tran.successful?
+      end
+    end
+     successful_invoice_item.reduce(0) do |sum, inv_item|
+      sum += inv_item.revenue
+    end.to_digits
```

Comment:
`end` at 45, 4 is not aligned with `successful_invoice_item.reduce(0) do |sum, inv_item|` at 43, 5